src/main/scala/parsley/Parsley.scala

```diff
@@ -1032,7 +1032,7 @@ private [parsley] object DeepEmbedding
             }
         override def findLetsAux[Cont[_, _]](implicit seen: Set[Parsley[_]], state: LetFinderState, ops: ContOps[Cont]): Cont[Unit, Unit] =
             for (_ <- _p.findLets; _ <- _q.findLets) yield ()
-        @tailrec override def optimise: Parsley[B] = p match
+        /* @tailrec */ override def optimise: Parsley[B] = p match
```

Comment:
Dotty complained about referring to superclasses.  This is a concern worth revisiting.

It shouldn't affect Scala 2: this annotation is just an assertion.  It's regrettable to lose it.

Reply:
That's actually vaguely annoying. I'll admit getting some of these `tailrec`s to fire relied on some pretty nefarious tricks to bypass the usual restrictions. I should take a closer look at this and make sure it's not created too big of an opportunity for stack overflow.

Reply:
Ah yes, I know what this is. The call it refers to is actually not a tail-rec call, and it's been cast to trick the compiler into still generating the recursion in that instance, but allow the tailrec in the other instances. Really, the `tailrec` annotation desperately needs some improvement imo. Hopefully I can figure out how to coerce into ignoring it again (I have some horrible ideas...)

Reply:
Yeah this'll do it:

```
// This is a trick to get tail-calls to fire even in the presence of a legimate recursion
final private [parsley] def optimiseDefinitelyNotTailRec: Parsley[A] = optimise
```